assets/app/views/create.html

```diff
@@ -4,25 +4,30 @@
       <div class="row">
         <div class="col-md-12">
           <h1>Create from ...</h1>
-          <!-- TODO need validation on source input field -->
-
-              <form name="from_source_form" class="form-inline gutter-bottom">
-              <h2>Source repository</h2>
-              <div class="row">
-                <div class="col-md-8">
-                  <div class="input-group">
-                    <span ng-class="{'has-error': from_source_form.from_source_url.$error.pattern}">
-                      <input class="form-control input-lg" name="from_source_url" placeholder="URL for source repository" type="text" required ng-model="from_source_url" ng-pattern="sourceURLPattern" ng-model-options="{ updateOn: 'blur' }"></input>
-                    </span>
-                    <span class="input-group-btn">
-                      <button type="submit" class="btn btn-lg btn-default" ng-click="createFromSource()" ng-enabled="from_source_form.$valid" value="">Next <i class="fa fa-arrow-right"></i>
-                      </button>
-                    </span>
-                  </div>
-                  <span class="text-danger" ng-if="from_source_form.from_source_url.$error.pattern">Please enter a valid URL</span>
+          <form name="from_source_form" class="form-inline gutter-bottom">
+            <h2>Source repository</h2>
+            <div class="row">
+              <div class="col-md-8">
+                <div class="input-group">
+                  <span ng-class="{'has-error': from_source_form.$invalid && from_source_form.from_source_url.$touched}">
+                    <input class="form-control input-lg"
+                      name="from_source_url"
+                      placeholder="URL for source repository"
+                      pattern="(((ssh|http(s)?)://)|(git(@[\w\.]+)?:)).+"
```

Comment:
There's an indication of an invalid value when you blur the field.
It is case sensitive.